### PR TITLE
#15371 Return app config by domain if app id isn't specified

### PR DESCRIPTION
**Pull request by i1shilyaev**

### CODE REVIEW

**Review by kshmidman — DISMISSED** [review]

File: src/main/java/io/kontur/disasterninja/controller/AppsController.java [hunk 1]
```diff
@@ -126,14 +127,20 @@ public List<LayerDetailsDto> updateListOfLayers(@PathVariable("id") UUID appId,
...
-            "appId is provided and requester domain is unknown", tags = {"Applications"})
+    @Operation(summary = "Get application config with features and user settings by id. Returns default app if " +
+            "no appId is provided and requester domain is unknown", tags = {"Applications"})
     @ApiResponse(responseCode = "200",
             content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE,
                     schema = @Schema(implementation = AppDto.class)))
     @GetMapping(path = "/configuration")
     public AppDto getAppConfig(@RequestParam(name = "appId", required = false) UUID appId,
-                               @RequestHeader(name = "Host", required = false) String domain) {
+                               @RequestHeader(name = "referer", required = false) String referer) {
```
Comment (Komzpa) [inline src/main/java/io/kontur/disasterninja/controller/AppsController.java:137]: Referer is not hostname that your endpoint is at but rather the one that issues tge request, and can be skipped by the browser in certain conditions. Tge correct header is Host really. Why did you change to referer?

Comment (i1shilyaev) [reply src/main/java/io/kontur/disasterninja/controller/AppsController.java:137]: Because `Host` header contains inner k8s link instead of the initial hostname. https://konturio.slack.com/archives/C037ZB1Q53P/p1683712911568459?thread_ts=1683712631.715549&cid=C037ZB1Q53P
If not using `Referer` header, the option that I see is for FE to pass the desired value in additional parameter or in another header.

Comment (Komzpa) [reply src/main/java/io/kontur/disasterninja/controller/AppsController.java:137]: Okay, if you have proxy chains that can't set Host properly (why? nginx can) then at least use X-Forwarded-Host https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Host

Comment (i1shilyaev) [reply src/main/java/io/kontur/disasterninja/controller/AppsController.java:137]: Sure. Additional changes are required then, probably in nginx config.

**Review by Komzpa — APPROVED** [review]
